compiler/backend/src/org/jetbrains/kotlin/codegen/inline/MethodInliner.kt

```diff
@@ -413,9 +413,18 @@ class MethodInliner(
 
         val capturedParamsSize = parameters.capturedParametersSizeOnStack
         val realParametersSize = parameters.realParametersSizeOnStack
+        val newArgumentList = if (!inliningContext.isInliningIrLambda) {
+            // Add captured parameters at the end to be used instead of the receiver.
+            Type.getArgumentTypes(node.desc) + parameters.capturedTypes
+        } else {
+            // Already have parameters for captured values, but they are actually at the beginning.
+            // The order of arguments in IR inline lambdas is extension?, captured*, real*;
```

Comment:
I would prefer to change order of parameters in IR because 'extension?, captured*, real*' looks too weird

Reply:
The problem is that real parameters can have default values:

```kotlin
fun f(x: Int) {
    fun g(y: Int = x) = y
    return g()
}
```

->

```kotlin
fun `f$g`(`$x`: Int, y: Int = `$x`) = y
fun f(x: Int) = `f$g`(x)
```

Swapping the order of arguments would result in invalid IR.